service/controller/drainer_resource_set.go

```diff
@@ -0,0 +1,95 @@
+package controller
+
+import (
+	"time"
+
+	"github.com/giantswarm/certs/v3/pkg/certs"
+	"github.com/giantswarm/k8sclient/v5/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/v4/pkg/resource"
+	"github.com/giantswarm/operatorkit/v4/pkg/resource/wrapper/metricsresource"
+	"github.com/giantswarm/operatorkit/v4/pkg/resource/wrapper/retryresource"
+	"github.com/giantswarm/tenantcluster/v4/pkg/tenantcluster"
+
+	"github.com/giantswarm/node-operator/service/controller/resource/drainer"
+)
+
+type DrainerResourceSetConfig struct {
+	K8sClient k8sclient.Interface
+	Logger    micrologger.Logger
+}
+
+func NewDrainerResourceSet(config DrainerResourceSetConfig) ([]resource.Interface, error) {
```

Comment:
Applying this suggestion would require more corresponding changes of course. We've gotten rid of the concept of "resource sets" so it's a chance to drop this terminology if you're up for it.
```suggestion
func NewDrainerResources(config DrainerConfig) ([]resource.Interface, error) {
```

Reply:
I think doing this in a follow up would be okay. So we get the first big PR out of restructuring and then clean up.